### PR TITLE
Add syntax highlighting for Julia

### PR DESCRIPTION
**Pull request by JobinsJC**

### CODE REVIEW

File: plugins/language_julia.lua [hunk 1]
```diff
@@ -0,0 +1,85 @@
+-- mod-version:2 -- lite-xl 2.0
+-- Support for the GDScript programming language: https://godotengine.org/
+-- Covers the most used keywords up to Godot version 3.2.x
```
Comment (Guldoman) [inline plugins/language_julia.lua:3]: Woops :P

File: plugins/language_julia.lua [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+local syntax = require "core.syntax"
+
+syntax.add {
+  files = { "%.jl$" },
```
Comment (Guldoman) [inline plugins/language_julia.lua:8]: Could you please add
```lua
name = "Julia",
```
here?

**Review by Guldoman — CHANGES_REQUESTED** [review]

File: plugins/language_julia.lua (hunk 2 above)
Comment (JobinsJC) [reply plugins/language_julia.lua:9]: Oops. Thanks

File: plugins/language_rust.lua [hunk 3]
```diff
@@ -15,6 +15,7 @@ syntax.add {
     { pattern = "-?%d+[%d%.eE]*f?",     type = "number"   },
     { pattern = "-?%.?%d+f?",           type = "number"   },
     { pattern = "[%+%-=/%*%^%%<>!~|&]", type = "operator" },
+    { pattern = "[%a_][%w_]*!%f[%[(]",  type = "function" },
```
Comment (Guldoman) [inline plugins/language_rust.lua:18]: Is this change to the Rust plugin intended?

Comment (JobinsJC) [reply plugins/language_rust.lua:18]: Yes. It is for highlighting macros. https://github.com/rxi/lite-plugins/pull/128

Comment (Guldoman) [reply plugins/language_rust.lua:18]: As this was about the Julia plugin it seemed a little strange, but that's ok.

**Pull request merged**